src/metaopt/io/database/mongodb.py

```diff
@@ -88,26 +90,16 @@ def unlock(self):
             raise_from(DatabaseError("Could not determine unlock due to connection."), e)
         return unlock
 
-    def insert(self, collection_name, document_dictionaries):
-        """ Inserts a document/multiple documents
-
-        Keyword arguments:
-        collection_name -- the collection name
-        document_dictionaries -- the dictionary/dictionaries that would be inserted
-                                 inside the collection
-        Returns:
-        result -- an instance of InsertManyResult which has inserted_ids and acknowledged
-        """
-
-        dbcollection = self.db[collection_name]
-        result = dbcollection.insert_many(document_dictionaries)
-
-        return result
-
-    def update(self, collection_name, query, update_dictionary,
-               selection, upsert=False, sort=None,
-               return_document=ReturnDocument.BEFORE):
-        """ Finds a document and modify it
+    def write(self, collection_name,
+                    query,
+                    dictionary,
+                    selection,
+                    atomic=False,
+                    upsert=False,
```

Comment:
If we always set it to `True` we'd lose power over just updating or doing an insert.

Reply:
No, upsert only inserts if the document is not found using its id. A very common use case will be to write new documents so it would be weird to always have to set database.write(new_documents, update=True). Also, in our case updates will usually be done by the objects representing the rows (Experiments, Trials, etc) so it will be simple to pass to `write()` the id so that `write()` does an update rather than an insert. All of that without specifying `upsert`.